agent/agent_endpoint.go

```diff
@@ -511,6 +511,57 @@ func (s *HTTPServer) AgentCheckUpdate(resp http.ResponseWriter, req *http.Reques
 	return nil, nil
 }
 
+func (s *HTTPServer) AgentHealthService(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	if req.Method != "GET" {
+		return nil, MethodNotAllowedError{req.Method, []string{"GET"}}
+	}
+
+	// Pull out the service id (service id since there may be several instance of the same service on this host)
+	serviceID := strings.TrimPrefix(req.URL.Path, "/v1/agent/health/service/")
+	if serviceID == "" {
+		resp.WriteHeader(http.StatusBadRequest)
+		fmt.Fprint(resp, "Missing service id")
+		return nil, nil
+	}
+
+	checks := s.agent.state.Checks()
+	// TODO: should we filter using acls like in AgentChecks method?
+	serviceChecks := make(api.HealthChecks, 0)
+	for _, c := range checks {
+		if c.ServiceID == serviceID {
```

Comment:
Usually these APIs are based on the service names.
It would be probably a good thing to also authorize calling the endpoint by passing the name. This would imply returning all the services matching the service name on the agent.

Reply:
Unless we add another endpoint, I think it might be confusing for users to interpret parameter as service id or name depending on xxx (based on what btw?).

Any input from hashicorp member?

Reply:
No, the endpoints could be different (/service vs /services for ex).
My point was to allow using svc names when using this API.